File: das_backend/src/test/java/ch/sbb/backend/arch/ModularityTests.java
package ch.sbb.backend.arch;

import ch.sbb.backend.DASBackendApplication;
import org.junit.jupiter.api.Test;
import org.springframework.modulith.core.ApplicationModules;

class ModularityTests {

    @Test
    void verifiesModularStructure() {
        ApplicationModules modules = ApplicationModules.of(DASBackendApplication.class);
        modules.forEach(System.out::println);
        modules.verify();
    }
}

File: backend/src/test/kotlin/ch/sbb/backend/ddd/ModularityTests.kt
package ch.sbb.backend.ddd

import ch.sbb.backend.BackendApplication
import org.junit.jupiter.api.Test
import org.springframework.modulith.core.ApplicationModules

class ModularityTests {

    @Test
    fun verifiesModularStructure() {
        val modules: ApplicationModules = ApplicationModules.of(BackendApplication::class.java)
        modules.forEach { println(it) }
        modules.verify()
    }
}
